<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Carl Boettiger" />
<meta name="author" content="Scott Chamberlain" />
<meta name="author" content="Rutger Vos" />
<meta name="author" content="Hilmar Lapp" />


<title>Handling Metadata in RNeXML</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; }
code > span.dt { color: #902000; }
code > span.dv { color: #40a070; }
code > span.bn { color: #40a070; }
code > span.fl { color: #40a070; }
code > span.ch { color: #4070a0; }
code > span.st { color: #4070a0; }
code > span.co { color: #60a0b0; font-style: italic; }
code > span.ot { color: #007020; }
code > span.al { color: #ff0000; font-weight: bold; }
code > span.fu { color: #06287e; }
code > span.er { color: #ff0000; font-weight: bold; }
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<link href="data:text/css,body%20%7B%0A%20%20background%2Dcolor%3A%20%23fff%3B%0A%20%20margin%3A%201em%20auto%3B%0A%20%20max%2Dwidth%3A%20700px%3B%0A%20%20overflow%3A%20visible%3B%0A%20%20padding%2Dleft%3A%202em%3B%0A%20%20padding%2Dright%3A%202em%3B%0A%20%20font%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0A%20%20font%2Dsize%3A%2014px%3B%0A%20%20line%2Dheight%3A%201%2E35%3B%0A%7D%0A%0A%23header%20%7B%0A%20%20text%2Dalign%3A%20center%3B%0A%7D%0A%0A%23TOC%20%7B%0A%20%20clear%3A%20both%3B%0A%20%20margin%3A%200%200%2010px%2010px%3B%0A%20%20padding%3A%204px%3B%0A%20%20width%3A%20400px%3B%0A%20%20border%3A%201px%20solid%20%23CCCCCC%3B%0A%20%20border%2Dradius%3A%205px%3B%0A%0A%20%20background%2Dcolor%3A%20%23f6f6f6%3B%0A%20%20font%2Dsize%3A%2013px%3B%0A%20%20line%2Dheight%3A%201%2E3%3B%0A%7D%0A%20%20%23TOC%20%2Etoctitle%20%7B%0A%20%20%20%20font%2Dweight%3A%20bold%3B%0A%20%20%20%20font%2Dsize%3A%2015px%3B%0A%20%20%20%20margin%2Dleft%3A%205px%3B%0A%20%20%7D%0A%0A%20%20%23TOC%20ul%20%7B%0A%20%20%20%20padding%2Dleft%3A%2040px%3B%0A%20%20%20%20margin%2Dleft%3A%20%2D1%2E5em%3B%0A%20%20%20%20margin%2Dtop%3A%205px%3B%0A%20%20%20%20margin%2Dbottom%3A%205px%3B%0A%20%20%7D%0A%20%20%23TOC%20ul%20ul%20%7B%0A%20%20%20%20margin%2Dleft%3A%20%2D2em%3B%0A%20%20%7D%0A%20%20%23TOC%20li%20%7B%0A%20%20%20%20line%2Dheight%3A%2016px%3B%0A%20%20%7D%0A%0Atable%20%7B%0A%20%20margin%3A%201em%20auto%3B%0A%20%20border%2Dwidth%3A%201px%3B%0A%20%20border%2Dcolor%3A%20%23DDDDDD%3B%0A%20%20border%2Dstyle%3A%20outset%3B%0A%20%20border%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0A%20%20border%2Dwidth%3A%202px%3B%0A%20%20padding%3A%205px%3B%0A%20%20border%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0A%20%20border%2Dwidth%3A%201px%3B%0A%20%20border%2Dstyle%3A%20inset%3B%0A%20%20line%2Dheight%3A%2018px%3B%0A%20%20padding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0A%20%20border%2Dleft%2Dstyle%3A%20none%3B%0A%20%20border%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0A%20%20background%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0A%0Ap%20%7B%0A%20%20margin%3A%200%2E5em%200%3B%0A%7D%0A%0Ablockquote%20%7B%0A%20%20background%2Dcolor%3A%20%23f6f6f6%3B%0A%20%20padding%3A%200%2E25em%200%2E75em%3B%0A%7D%0A%0Ahr%20%7B%0A%20%20border%2Dstyle%3A%20solid%3B%0A%20%20border%3A%20none%3B%0A%20%20border%2Dtop%3A%201px%20solid%20%23777%3B%0A%20%20margin%3A%2028px%200%3B%0A%7D%0A%0Adl%20%7B%0A%20%20margin%2Dleft%3A%200%3B%0A%7D%0A%20%20dl%20dd%20%7B%0A%20%20%20%20margin%2Dbottom%3A%2013px%3B%0A%20%20%20%20margin%2Dleft%3A%2013px%3B%0A%20%20%7D%0A%20%20dl%20dt%20%7B%0A%20%20%20%20font%2Dweight%3A%20bold%3B%0A%20%20%7D%0A%0Aul%20%7B%0A%20%20margin%2Dtop%3A%200%3B%0A%7D%0A%20%20ul%20li%20%7B%0A%20%20%20%20list%2Dstyle%3A%20circle%20outside%3B%0A%20%20%7D%0A%20%20ul%20ul%20%7B%0A%20%20%20%20margin%2Dbottom%3A%200%3B%0A%20%20%7D%0A%0Apre%2C%20code%20%7B%0A%20%20background%2Dcolor%3A%20%23f7f7f7%3B%0A%20%20border%2Dradius%3A%203px%3B%0A%20%20color%3A%20%23333%3B%0A%7D%0Apre%20%7B%0A%20%20white%2Dspace%3A%20pre%2Dwrap%3B%20%20%20%20%2F%2A%20Wrap%20long%20lines%20%2A%2F%0A%20%20border%2Dradius%3A%203px%3B%0A%20%20margin%3A%205px%200px%2010px%200px%3B%0A%20%20padding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0A%20%20background%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0A%0Acode%20%7B%0A%20%20font%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0A%20%20font%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0A%20%20padding%3A%202px%200px%3B%0A%7D%0A%0Adiv%2Efigure%20%7B%0A%20%20text%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0A%20%20background%2Dcolor%3A%20%23FFFFFF%3B%0A%20%20padding%3A%202px%3B%0A%20%20border%3A%201px%20solid%20%23DDDDDD%3B%0A%20%20border%2Dradius%3A%203px%3B%0A%20%20border%3A%201px%20solid%20%23CCCCCC%3B%0A%20%20margin%3A%200%205px%3B%0A%7D%0A%0Ah1%20%7B%0A%20%20margin%2Dtop%3A%200%3B%0A%20%20font%2Dsize%3A%2035px%3B%0A%20%20line%2Dheight%3A%2040px%3B%0A%7D%0A%0Ah2%20%7B%0A%20%20border%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0A%20%20padding%2Dtop%3A%2010px%3B%0A%20%20padding%2Dbottom%3A%202px%3B%0A%20%20font%2Dsize%3A%20145%25%3B%0A%7D%0A%0Ah3%20%7B%0A%20%20border%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0A%20%20padding%2Dtop%3A%2010px%3B%0A%20%20font%2Dsize%3A%20120%25%3B%0A%7D%0A%0Ah4%20%7B%0A%20%20border%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0A%20%20margin%2Dleft%3A%208px%3B%0A%20%20font%2Dsize%3A%20105%25%3B%0A%7D%0A%0Ah5%2C%20h6%20%7B%0A%20%20border%2Dbottom%3A%201px%20solid%20%23ccc%3B%0A%20%20font%2Dsize%3A%20105%25%3B%0A%7D%0A%0Aa%20%7B%0A%20%20color%3A%20%230033dd%3B%0A%20%20text%2Ddecoration%3A%20none%3B%0A%7D%0A%20%20a%3Ahover%20%7B%0A%20%20%20%20color%3A%20%236666ff%3B%20%7D%0A%20%20a%3Avisited%20%7B%0A%20%20%20%20color%3A%20%23800080%3B%20%7D%0A%20%20a%3Avisited%3Ahover%20%7B%0A%20%20%20%20color%3A%20%23BB00BB%3B%20%7D%0A%20%20a%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0A%20%20%20%20text%2Ddecoration%3A%20underline%3B%20%7D%0A%20%20a%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0A%20%20%20%20text%2Ddecoration%3A%20underline%3B%20%7D%0A%0A%2F%2A%20Class%20described%20in%20https%3A%2F%2Fbenjeffrey%2Ecom%2Fposts%2Fpandoc%2Dsyntax%2Dhighlighting%2Dcss%0A%20%20%20Colours%20from%20https%3A%2F%2Fgist%2Egithub%2Ecom%2Frobsimmons%2F1172277%20%2A%2F%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%2F%2A%20Keyword%20%2A%2F%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%2F%2A%20DataType%20%2A%2F%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%2F%2A%20DecVal%20%28decimal%20values%29%20%2A%2F%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20BaseN%20%2A%2F%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20Float%20%2A%2F%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20Char%20%2A%2F%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20String%20%2A%2F%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%2F%2A%20Comment%20%2A%2F%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%2F%2A%20OtherToken%20%2A%2F%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%2F%2A%20AlertToken%20%2A%2F%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%2F%2A%20Function%20calls%20%2A%2F%20%0Acode%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%2F%2A%20ErrorTok%20%2A%2F%0A%0A" rel="stylesheet" type="text/css" />

</head>

<body>



<div id="header">
<h1 class="title">Handling Metadata in RNeXML</h1>
<h4 class="author"><em>Carl Boettiger</em></h4>
<h4 class="author"><em>Scott Chamberlain</em></h4>
<h4 class="author"><em>Rutger Vos</em></h4>
<h4 class="author"><em>Hilmar Lapp</em></h4>
</div>


<!--
%\VignetteEngine{knitr::rmarkdown}
%\VignetteIndexEntry{Handling Metadata in RNeXML}
-->
<div id="writing-nexml-metadata" class="section level2">
<h2>Writing NeXML metadata</h2>
<p>The <code>add_basic_meta()</code> function takes as input an existing <code>nexml</code> object (like the other <code>add_</code> functions, if none is provided it will create one), and at the time of this writing any of the following parameters: <code>title</code>, <code>description</code>, <code>creator</code>, <code>pubdate</code>, <code>rights</code>, <code>publisher</code>, <code>citation</code>. Other metadata elements and corresponding parameters may be added in the future.</p>
<p>Load the packages and data:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">'RNeXML'</span>)
<span class="kw">data</span>(bird.orders)</code></pre>
<p>Create an <code>nexml</code> object for the phylogeny <code>bird.orders</code> and add appropriate metadata:</p>
<pre class="sourceCode r"><code class="sourceCode r">birds &lt;-<span class="st"> </span><span class="kw">add_trees</span>(bird.orders)
birds &lt;-<span class="st"> </span><span class="kw">add_basic_meta</span>(
  <span class="dt">title =</span> <span class="st">&quot;Phylogeny of the Orders of Birds From Sibley and Ahlquist&quot;</span>,

  <span class="dt">description =</span> <span class="st">&quot;This data set describes the phylogenetic relationships of the</span>
<span class="st">     orders of birds as reported by Sibley and Ahlquist (1990). Sibley</span>
<span class="st">     and Ahlquist inferred this phylogeny from an extensive number of</span>
<span class="st">     DNA/DNA hybridization experiments. The ``tapestry'' reported by</span>
<span class="st">     these two authors (more than 1000 species out of the ca. 9000</span>
<span class="st">     extant bird species) generated a lot of debates.</span>

<span class="st">     The present tree is based on the relationships among orders. The</span>
<span class="st">     branch lengths were calculated from the values of Delta T50H as</span>
<span class="st">     found in Sibley and Ahlquist (1990, fig. 353).&quot;</span>,

  <span class="dt">citation =</span> <span class="st">&quot;Sibley, C. G. and Ahlquist, J. E. (1990) Phylogeny and</span>
<span class="st">     classification of birds: a study in molecular evolution. New</span>
<span class="st">     Haven: Yale University Press.&quot;</span>,

  <span class="dt">creator =</span> <span class="st">&quot;Sibley, C. G. and Ahlquist, J. E.&quot;</span>,
    <span class="dt">nexml=</span>birds)</code></pre>
<p>Instead of a literal string, citations can also be provided in R’s <code>bibentry</code> type, which is the one in which R package citations are obtained:</p>
<pre class="sourceCode r"><code class="sourceCode r">birds &lt;-<span class="st"> </span><span class="kw">add_basic_meta</span>(<span class="dt">citation =</span> <span class="kw">citation</span>(<span class="st">&quot;ape&quot;</span>), <span class="dt">nexml =</span> birds)</code></pre>
</div>
<div id="taxonomic-identifiers" class="section level2">
<h2>Taxonomic identifiers</h2>
<p>The <code>taxize_nexml()</code> function uses the R package <code>taxize</code> <span class="citation">[@Chamberlain_2013]</span> to check each taxon label against the NCBI database. If a unique match is found, a metadata annotation is added to the taxon providing the NCBI identification number to the taxonomic unit.</p>
<pre class="sourceCode r"><code class="sourceCode r">birds &lt;-<span class="st"> </span><span class="kw">taxize_nexml</span>(birds, <span class="st">&quot;NCBI&quot;</span>)</code></pre>
<p>If no match is found, the user is warned to check for possible typographic errors in the taxonomic labels provided. If multiple matches are found, the user will be prompted to choose between them.</p>
</div>
<div id="custom-metadata-extensions" class="section level2">
<h2>Custom metadata extensions</h2>
<p>We can get a list of namespaces along with their prefixes from the <code>nexml</code> object:</p>
<pre class="sourceCode r"><code class="sourceCode r">prefixes &lt;-<span class="st"> </span><span class="kw">get_namespaces</span>(birds)
prefixes[<span class="st">&quot;dc&quot;</span>]</code></pre>
<pre><code>                                dc 
&quot;http://purl.org/dc/elements/1.1/&quot; </code></pre>
<p>We create a <code>meta</code> element containing this annotation using the <code>meta</code> function:</p>
<pre class="sourceCode r"><code class="sourceCode r">modified &lt;-<span class="st"> </span><span class="kw">meta</span>(<span class="dt">property =</span> <span class="st">&quot;prism:modificationDate&quot;</span>, <span class="dt">content =</span> <span class="st">&quot;2013-10-04&quot;</span>)</code></pre>
<p>We can add this annotation to our existing <code>birds</code> NeXML file using the <code>add_meta()</code> function. Because we do not specify a level, it is added to the root node, referring to the NeXML file as a whole.</p>
<pre class="sourceCode r"><code class="sourceCode r">birds &lt;-<span class="st"> </span><span class="kw">add_meta</span>(modified, birds) </code></pre>
<p>The built-in vocabularies are just the tip of the iceberg of established vocabularies. Here we add an annotation from the <code>skos</code> namespace which describes the history of where the data comes from:</p>
<pre class="sourceCode r"><code class="sourceCode r">history &lt;-<span class="st"> </span><span class="kw">meta</span>(<span class="dt">property =</span> <span class="st">&quot;skos:historyNote&quot;</span>,
  <span class="dt">content =</span> <span class="st">&quot;Mapped from the bird.orders data in the ape package using RNeXML&quot;</span>)</code></pre>
<p>Because <code>skos</code> is not in the current namespace list, we add it with a url when adding this meta element. We also specify that this annotation be placed at the level of the <code>trees</code> sub-node in the NeXML file.</p>
<pre class="sourceCode r"><code class="sourceCode r">birds &lt;-<span class="st"> </span><span class="kw">add_meta</span>(history, 
                birds, 
                <span class="dt">level =</span> <span class="st">&quot;trees&quot;</span>,
                <span class="dt">namespaces =</span> <span class="kw">c</span>(<span class="dt">skos =</span> <span class="st">&quot;http://www.w3.org/2004/02/skos/core#&quot;</span>))</code></pre>
<p>For finer control of the level at which a <code>meta</code> element is added, we will manipulate the <code>nexml</code> R object directly using S4 sub-setting, as shown in the supplement.</p>
<p>Much richer metadata annotation is possible. Later we illustrate how metadata annotation can be used to extend the base NeXML format to represent new forms of data while maintaining compatibility with any NeXML parser. The <code>RNeXML</code> package can be easily extended to support helper functions such as <code>taxize_nexml</code> to add additional metadata without imposing a large burden on the user.</p>
</div>
<div id="reading-nexml-metadata" class="section level2">
<h2>Reading NeXML metadata</h2>
<p>A call to the <code>nexml</code> object prints some metadata summarizing the data structure:</p>
<pre class="sourceCode r"><code class="sourceCode r">birds</code></pre>
<pre><code>A nexml object representing:
     1 phylogenetic tree blocks, where: 
     block 1 contains 1 phylogenetic trees 
     46 meta elements 
     0 character matrices 
     23 taxonomic units 
 Taxa:   Struthioniformes, Tinamiformes, Craciformes, Galliformes, Anseriformes, Turniciformes ... 

 NeXML generated by RNeXML using schema version: 0.9 
 size: 372.6 Kb </code></pre>
<p>We can extract all metadata pertaining to the NeXML document as a whole (annotations of the XML root node, <code>&lt;nexml&gt;</code>) with the command</p>
<pre class="sourceCode r"><code class="sourceCode r">meta &lt;-<span class="st"> </span><span class="kw">get_metadata</span>(birds) </code></pre>
<p>This returns a named list of available metadata. We can see the kinds of metadata recorded from the names (showing the first 4):</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">names</span>(meta)[<span class="dv">1</span>:<span class="dv">4</span>]</code></pre>
<pre><code>[1] &quot;dc:title&quot;                      &quot;dc:creator&quot;                   
[3] &quot;dc:description&quot;                &quot;dcterms:bibliographicCitation&quot;</code></pre>
<p>and can ask for a particular element using the standard list sub-setting mechanism (i.e. either the name of an element or its numeric position),</p>
<pre class="sourceCode r"><code class="sourceCode r">meta[[<span class="st">&quot;dc:title&quot;</span>]]</code></pre>
<pre><code>[1] &quot;Phylogeny of the Orders of Birds From Sibley and Ahlquist&quot;</code></pre>
<p>All metadata terms must belong to an explicit <em>namespace</em> or vocabulary that allows a computer to interpret the term precisely. The prefix (before the <code>:</code>) indicates to which vocabulary the term belongs, e.g. <code>dc</code> in this case. The <code>get_namespaces</code> function tells us the definition of the vocabulary using a link:</p>
<pre class="sourceCode r"><code class="sourceCode r">prefixes &lt;-<span class="st"> </span><span class="kw">get_namespaces</span>(birds)
prefixes[<span class="st">&quot;dc&quot;</span>]</code></pre>
<pre><code>                                dc 
&quot;http://purl.org/dc/elements/1.1/&quot; </code></pre>
<p>Common metadata can be accessed with a few dedicated functions:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">get_citation</span>(birds)</code></pre>
<pre><code>Sibley, C. G. and Ahlquist, J. E. (1990) Phylogeny and
     classification of birds: a study in molecular evolution. New
     Haven: Yale University Press. Paradis E, Claude J and Strimmer K (2004). &quot;APE: analyses of
phylogenetics and evolution in R language.&quot; _Bioinformatics_,
*20*, pp. 289-290.</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">get_taxa</span>(birds)</code></pre>
<pre><code> [1] &quot;Struthioniformes&quot; &quot;Tinamiformes&quot;     &quot;Craciformes&quot;     
 [4] &quot;Galliformes&quot;      &quot;Anseriformes&quot;     &quot;Turniciformes&quot;   
 [7] &quot;Piciformes&quot;       &quot;Galbuliformes&quot;    &quot;Bucerotiformes&quot;  
[10] &quot;Upupiformes&quot;      &quot;Trogoniformes&quot;    &quot;Coraciiformes&quot;   
[13] &quot;Coliiformes&quot;      &quot;Cuculiformes&quot;     &quot;Psittaciformes&quot;  
[16] &quot;Apodiformes&quot;      &quot;Trochiliformes&quot;   &quot;Musophagiformes&quot; 
[19] &quot;Strigiformes&quot;     &quot;Columbiformes&quot;    &quot;Gruiformes&quot;      
[22] &quot;Ciconiiformes&quot;    &quot;Passeriformes&quot;   </code></pre>
<p>Which returns text from the otu element labels, typically used to define taxonomic names, rather than text from explicit meta elements.</p>
<p>We can also access metadata at a specific level (or use <code>level=all</code> to extract all meta elements in a list). Here we show only the first few results:</p>
<pre class="sourceCode r"><code class="sourceCode r">otu_meta &lt;-<span class="st"> </span><span class="kw">get_metadata</span>(birds, <span class="dt">level=</span><span class="st">&quot;otu&quot;</span>)
otu_meta[<span class="dv">1</span>:<span class="dv">4</span>]</code></pre>
<pre><code>$`tc:toTaxon`
[1] &quot;http://ncbi.nlm.nih.gov/taxonomy/8798&quot;

$`tc:toTaxon`
[1] &quot;http://ncbi.nlm.nih.gov/taxonomy/8802&quot;

$`tc:toTaxon`
[1] &quot;http://ncbi.nlm.nih.gov/taxonomy/8976&quot;

$`tc:toTaxon`
[1] &quot;http://ncbi.nlm.nih.gov/taxonomy/8976&quot;</code></pre>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
